### PR TITLE
Multi modal chat

### PR DESCRIPTION
**Pull request by Mukamik**
Adding the ability to upload images to chatbot messages.
TODO to make the image section of the message optional.

### CODE REVIEW

File: src/frontend/handlers.go [hunk 1]
```diff
@@ -425,6 +424,7 @@ func (fe *frontendServer) logoutHandler(w http.ResponseWriter, r *http.Request)
 func chatBotHandler(w http.ResponseWriter, r *http.Request) {
 	type message struct {
 		Message string `json:"message"`
+		image   string `json:"image"`
```
Comment (moficodes) [inline src/frontend/handlers.go:427]: Uppercase to be exported

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,122 @@
+#
```
Comment (bourgeoisor) [inline requirements.txt:1]: Is this file meant to be checked in at the root of the repo? Shouldn't it be in `src/` somewhere?

File: kubernetes-manifests/shoppingassistantservice.yaml [hunk 3]
```diff
@@ -42,7 +42,7 @@ spec:
             drop:
               - ALL
           privileged: false
-          readOnlyRootFilesystem: true
+          readOnlyRootFilesystem: false
```
Comment (bourgeoisor) [inline kubernetes-manifests/shoppingassistantservice.yaml:45]: Why is this necessary? Ideally we don't need root access in our container as it's a security risk. None of the other services in Online Boutique needs it, so I'm curious what makes this one different?

File: src/shoppingassistantservice/requirements.txt [hunk 4]
```diff
@@ -4,80 +4,187 @@
 #
 #    pip-compile --output-file=requirements.txt requirements.in
 #
+aiohttp==3.9.3
+    # via
+<<<<<<< Updated upstream
```
Comment (bourgeoisor) [inline src/shoppingassistantservice/requirements.txt:9]: Looks like git merge artifacts got checked-in

File: src/shoppingassistantservice/shoppingassistantservice.py [hunk 5]
```diff
@@ -14,22 +14,112 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import os
 from urllib.parse import unquote
+from langchain_core.messages import HumanMessage
+<<<<<<< Updated upstream
```
Comment (bourgeoisor) [inline src/shoppingassistantservice/shoppingassistantservice.py:20]: More git merge artifacts in this file too

File: kubernetes-manifests/shoppingassistantservice.yaml (hunk 3 above)
Comment (Mukamik) [reply kubernetes-manifests/shoppingassistantservice.yaml:45]: It was a result of a failure to write to the tmp directory when accessing alloyDB. I tried granting permissions and defining a tmp directory that wasn't on the root to get around this, but it still failed to write, so I took this workaround for the sake of time.

File: scripts/1_deploy_alloydb_infra.sh [hunk 6]
```diff
@@ -4,11 +4,9 @@ set -e
...
 # Replace me
-PROJECT_ID=<project_id>
-PROJECT_NUMBER=<project_number>
-PGPASSWORD=<password>
-
-# Set sensible defaults
+PROJECT_ID=cooking-with-duet-6
+PROJECT_NUMBER=925071387448
+PGPASSWORD=admin
 REGION=us-central1
```
Comment (bourgeoisor) [inline scripts/1_deploy_alloydb_infra.sh:10]:
```suggestion
PROJECT_ID=<project_id>
PROJECT_NUMBER=<project_number>
PGPASSWORD=<password>

# Set sensible defaults
REGION=us-central1
```

File: kustomize/components/alloydb/kustomization.yaml [hunk 7]
```diff
@@ -32,23 +32,30 @@ patches:
...
       annotations:
-        iam.gke.io/gcp-service-account: ALLOYDB_USER_GSA_ID
+        iam.gke.io/gcp-service-account: alloydb-user-sa@cooking-with-duet-6.iam.gserviceaccount.com
+- patch: |-
+    apiVersion: v1
+    kind: ServiceAccount
+    metadata:
+      name: shoppingassistantservice
+      annotations:
+        iam.gke.io/gcp-service-account: alloydb-user-sa@cooking-with-duet-6.iam.gserviceaccount.com
```
Comment (bourgeoisor) [inline kustomize/components/alloydb/kustomization.yaml:58]:
```suggestion
        iam.gke.io/gcp-service-account: ALLOYDB_USER_GSA_ID```

File: kustomize/components/alloydb/kustomization.yaml [hunk 8]
```diff
@@ -32,23 +32,30 @@ patches:
...
+                value: carts
               - name: ALLOYDB_TABLE_NAME
-                value: ALLOYDB_TABLE_NAME_VAL
+                value: cart_items
               - name: ALLOYDB_SECRET_NAME
-                value: ALLOYDB_SECRET_NAME_VAL
+                value: alloydb-secret
               - name: PROJECT_ID
-                value: PROJECT_ID_VAL
+                value: cooking-with-duet-6
```
Comment (bourgeoisor) [inline kustomize/components/alloydb/kustomization.yaml:43]:
```suggestion
                value: PROJECT_ID_VAL
```

File: kustomize/components/alloydb/kustomization.yaml [hunk 9]
```diff
@@ -32,23 +32,30 @@ patches:
...
+                value: 10.39.0.2
               - name: ALLOYDB_DATABASE_NAME
-                value: ALLOYDB_DATABASE_NAME_VAL
+                value: carts
               - name: ALLOYDB_TABLE_NAME
-                value: ALLOYDB_TABLE_NAME_VAL
+                value: cart_items
               - name: ALLOYDB_SECRET_NAME
-                value: ALLOYDB_SECRET_NAME_VAL
+                value: alloydb-secret
```
Comment (bourgeoisor) [inline kustomize/components/alloydb/kustomization.yaml:41]:
```suggestion
                value: ALLOYDB_SECRET_NAME_VAL
```

File: kustomize/components/alloydb/kustomization.yaml [hunk 10]
```diff
@@ -32,23 +32,30 @@ patches:
...
                 $patch: delete
               - name: ALLOYDB_PRIMARY_IP
-                value: ALLOYDB_PRIMARY_IP_VAL
+                value: 10.39.0.2
               - name: ALLOYDB_DATABASE_NAME
-                value: ALLOYDB_DATABASE_NAME_VAL
+                value: carts
               - name: ALLOYDB_TABLE_NAME
-                value: ALLOYDB_TABLE_NAME_VAL
+                value: cart_items
```
Comment (bourgeoisor) [inline kustomize/components/alloydb/kustomization.yaml:39]:
```suggestion
                value: ALLOYDB_TABLE_NAME_VAL
```

File: kustomize/components/alloydb/kustomization.yaml [hunk 11]
```diff
@@ -32,23 +32,30 @@ patches:
               - name: REDIS_ADDR
                 $patch: delete
               - name: ALLOYDB_PRIMARY_IP
-                value: ALLOYDB_PRIMARY_IP_VAL
+                value: 10.39.0.2
               - name: ALLOYDB_DATABASE_NAME
-                value: ALLOYDB_DATABASE_NAME_VAL
+                value: carts
```
Comment (bourgeoisor) [inline kustomize/components/alloydb/kustomization.yaml:37]:
```suggestion
                value: ALLOYDB_DATABASE_NAME_VAL
```

File: kustomize/components/alloydb/kustomization.yaml [hunk 12]
```diff
@@ -32,23 +32,30 @@ patches:
               - name: REDIS_ADDR
                 $patch: delete
               - name: ALLOYDB_PRIMARY_IP
-                value: ALLOYDB_PRIMARY_IP_VAL
+                value: 10.39.0.2
```
Comment (bourgeoisor) [inline kustomize/components/alloydb/kustomization.yaml:35]:
```suggestion
                value: ALLOYDB_PRIMARY_IP_VAL
```

**Review by bourgeoisor — APPROVED** [review]
